crates/cli-support/src/js/mod.rs

```diff
@@ -1164,6 +1165,124 @@ __wbg_set_wasm(wasm);"
         Ok(())
     }
 
+    fn write_class_field_types(&mut self, class: &ExportedClass, ts_dst: &mut String) {
+        let mut fields: Vec<&FieldInfo> = class.typescript_fields.values().collect();
+        fields.sort_by_key(|f| f.order); // make sure we have deterministic output
+
+        for FieldInfo {
+            name,
+            is_static,
+            getter,
+            setter,
+            ..
+        } in fields
+        {
+            let is_static = if *is_static { "static " } else { "" };
+
+            let write_docs = |ts_dst: &mut String, docs: &str| {
+                if docs.is_empty() {
+                    return;
+                }
+                // indent by 2 spaces
+                for line in docs.lines() {
+                    ts_dst.push_str("  ");
+                    ts_dst.push_str(line);
+                    ts_dst.push('\n');
+                }
+            };
+            let write_getter = |ts_dst: &mut String, getter: &FieldAccessor| {
+                write_docs(ts_dst, &getter.docs);
+                ts_dst.push_str("  ");
+                ts_dst.push_str(is_static);
+                ts_dst.push_str("get ");
+                ts_dst.push_str(name);
+                ts_dst.push_str("(): ");
+                ts_dst.push_str(&getter.ty);
+                ts_dst.push_str(";\n");
+            };
+            let write_setter = |ts_dst: &mut String, setter: &FieldAccessor| {
+                write_docs(ts_dst, &setter.docs);
+                ts_dst.push_str("  ");
+                ts_dst.push_str(is_static);
+                ts_dst.push_str("set ");
+                ts_dst.push_str(name);
+                ts_dst.push_str("(value: ");
+                ts_dst.push_str(&setter.ty);
+                if setter.is_optional {
+                    ts_dst.push_str(" | undefined");
+                }
+                ts_dst.push_str(");\n");
+            };
```

Comment:
None of those have to be closures, right?
How about methods on `FieldAccessor`?

Reply:
`write_[gs]etter` capture the local variable `is_static`, but I really just wanted to have the printing logic all in one place. I could make them methods of `FieldAccessor`, but that would spread out the printing logic over multiple functions and 2 types. I prefer to have related things close together.

Reply:
Ah, somehow missed that capture.

> Also, is there a reason not to use closures for helper functions?

No, I just find it harder to read.
My experience so far in Rust has been that most people tend to avoid closures when possible.

No worries though!